util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java

```diff
@@ -220,16 +228,20 @@ private boolean acceptResolvedAddressesInternal(ResolvedAddresses resolvedAddres
         childLbStates.get(key).deactivate();
       }
     }
-    // Must update channel picker before return so that new RPCs will not be routed to deleted
-    // clusters and resolver can remove them in service config.
-    updateOverallBalancingState();
+
+    if (updateBalancingState) {
+      // Must update channel picker before return so that new RPCs will not be routed to deleted
+      // clusters and resolver can remove them in service config.
+      updateOverallBalancingState();
```

Comment:
I see that we have removing/shutdown childLB first, then updateOverAllBalancingState, probably because the original ClusterManagerLB is in this order.

However, I don't see problem in flipping the order. In fact, the original RoundRobinLB policy suggest this order (see [first update, then remove](https://github.com/grpc/grpc-java/blob/master/util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java#L223))  And ringhash LB might copy from RR. By doing that, we can remove the boolean parameter.

Reply:
The boolean parameter is needed because we can't do the updateOverAllBalancingState until we've processed the weights which can't be done until the child list is updated.  
Currently disable both updates the child list and does shutdown.  I'll split the shutdown from disable so that it can be done after updateOverAllBalancingState, but it complicates the logic.  Also makes it easier for removed subchannels to never be shutdown, but that should show up pretty obviously in testing.

Reply:
I agree that updateOverallBalancingState should be after child list is updated. 
But processing the weights (build ring) can be before the child list is updated, right? 
Either way, there is a slight time window where the ring is updated but child list is using the old update. But this is an existing problem. To fix that I believe we should have the ring be built into the picker, not as a LB field.

I suggest that how about in MultiChildLoadBalancer we remove ` updateOverallBalancingState()` and let the child policy do the update. This will remove the parameter. It also helps to make sure the state(LB configs and child list ) is consistent. For example, I see that [wrr update list then update config looks not entirely correct](https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java#L122). The more correct way is to update child list first, do `afterAcceptAddress()`, then finally updateLoadBalancingState.

Reply:
I've split it out into a separate method `updateLbStateAndShutdownRemoved ` which allows subclasses to call it at the end of their processing.
The ring doesn't matter as picker creation takes a snapshot, so the old snapshot will be used until the new picker is created and `helper.updateBalancingState` is called.